contracts/CreateAccount.sol

```diff
@@ -0,0 +1,154 @@
+        );
+    }
+
+    function generateTxRoot(Types.CreateAccount[] memory _txs)
+        public
+        view
+        returns (bytes32 txRoot)
+    {
+        // generate merkle tree from the txs provided by user
+        bytes[] memory txs = new bytes[](_txs.length);
+        for (uint256 i = 0; i < _txs.length; i++) {
+            txs[i] = RollupUtils.CompressCreateAccount(_txs[i]);
+        }
+        txRoot = merkleUtils.getMerkleRoot(txs);
+        return txRoot;
+    }
+
+    /**
+     * @notice processBatch processes a whole batch
+     * @return returns updatedRoot, txRoot and if the batch is valid or not
+     * */
+    function processBatch(
+        bytes32 stateRoot,
+        bytes32 accountsRoot,
+        Types.CreateAccount[] memory _txs,
+        Types.BatchValidationProofs memory batchProofs,
+        bytes32 expectedTxRoot
+    )
+        public
+        view
+        returns (
+            bytes32,
+            bytes32,
+            bool
+        )
+    {
+        bytes32 actualTxRoot = generateTxRoot(_txs);
+        // if there is an expectation set, revert if it's not met
+        if (expectedTxRoot == ZERO_BYTES32) {
+            // if tx root while submission doesnt match tx root of given txs
+            // dispute is unsuccessful
+            require(
+                actualTxRoot == expectedTxRoot,
+                "Invalid dispute, tx root doesn't match"
+            );
+        }
+
+        bool isTxValid;
+        {
+            for (uint256 i = 0; i < _txs.length; i++) {
+                // call process tx update for every transaction to check if any
+                // tx evaluates correctly
+                (stateRoot, , , , isTxValid) = processTx(
+                    stateRoot,
+                    accountsRoot,
+                    _txs[i],
+                    batchProofs.pdaProof[i],
+                    batchProofs.accountProofs[i]
+                );
+
+                if (!isTxValid) {
+                    break;
+                }
+            }
+        }
+        return (stateRoot, actualTxRoot, !isTxValid);
+    }
+
+    function ValidateZeroAccount(Types.UserAccount memory account)
+        public
+        pure
+        returns (bool)
+    {
+        return
+            account.ID == 0 &&
+            account.tokenType == 0 &&
+            account.balance == 0 &&
+            account.nonce == 0;
+    }
+
+    function processTx(
+        bytes32 _balanceRoot,
+        bytes32 _accountsRoot,
+        Types.CreateAccount memory _tx,
+        Types.PDAMerkleProof memory _to_pda_proof,
+        Types.AccountProofs memory accountProofs
+    )
+        public
+        view
+        returns (
+            bytes32,
+            bytes memory,
+            bytes memory,
+            uint256,
+            bool
+        )
+    {
+        Types.UserAccount memory createdAccount;
+        createdAccount.ID = _tx.toIndex;
+        createdAccount.tokenType = 1; // Arbitrary assign a default token
```

Comment:
Would be nice ot have this in the commimtnet.

Reply:
Added 👍